File: sdks/csharp/src/NovaBilling/Taxes/Requests/UpdateTaxDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record UpdateTaxDto
{
    /// <summary>
    /// Tax ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    [JsonPropertyName("name")]
    public string? Name { get; set; }

    [JsonPropertyName("rate")]
    public double? Rate { get; set; }

    [JsonPropertyName("description")]
    public string? Description { get; set; }

    [JsonPropertyName("appliedByDefault")]
    public bool? AppliedByDefault { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Taxes/Requests/UpdateTaxDto.php
<?php

namespace NovaBilling\Taxes\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class UpdateTaxDto extends JsonSerializableType
{
    /**
     * @var ?string $name
     */
    #[JsonProperty('name')]
    public ?string $name;

    /**
     * @var ?float $rate
     */
    #[JsonProperty('rate')]
    public ?float $rate;

    /**
     * @var ?string $description
     */
    #[JsonProperty('description')]
    public ?string $description;

    /**
     * @var ?bool $appliedByDefault
     */
    #[JsonProperty('appliedByDefault')]
    public ?bool $appliedByDefault;

    /**
     * @param array{
     *   name?: ?string,
     *   rate?: ?float,
     *   description?: ?string,
     *   appliedByDefault?: ?bool,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->name = $values['name'] ?? null;
        $this->rate = $values['rate'] ?? null;
        $this->description = $values['description'] ?? null;
        $this->appliedByDefault = $values['appliedByDefault'] ?? null;
    }
}
